ansible/hosts.yaml

```diff
@@ -0,0 +1,206 @@
+      version: 4.13.40
+      network_prefix_cidr: 10.0.0.0/22
+      master_cidr: 10.0.0.0/23
+      master_size: Standard_E8s_v5
+      worker_cidr: 10.0.2.0/23
+      worker_size: Standard_D4s_v5
+
+encrypted_clusters:
+  # Basic cluster with encryption-at-host enabled
+  hosts:
+    enc413:
+      name: aro
+      version: 4.13.40
+    enc414:
+      name: aro
+      version: 4.14.16
+  vars:
+    resource_group: "{{ CLUSTERPREFIX }}-{{ inventory_hostname }}-{{ location }}"
+    network_prefix_cidr: 10.0.0.0/22
+    master_cidr: 10.0.0.0/23
+    worker_cidr: 10.0.2.0/23
+    master_size: Standard_E8s_v5
+    master_encryption_at_host: true
+    worker_size: Standard_D4s_v5
+    worker_encryption_at_host: true
+
+private_clusters:
+  hosts:
+    private:
+      # Simple private cluster, no UDR
+      name: aro
+      resource_group: "{{ CLUSTERPREFIX }}-private-{{ location }}"
+      apiserver_visibility: Private
+      ingress_visibility: Private
+      network_prefix_cidr: 10.0.0.0/22
+      master_cidr: 10.0.0.0/23
+      master_size: Standard_D8s_v3
+      worker_cidr: 10.0.2.0/23
+      worker_size: Standard_D4s_v3
+
+udr_clusters:
+  # https://learn.microsoft.com/en-us/azure/openshift/howto-create-private-cluster-4x
+  hosts:
+    udr414:
+      name: aro-414
+      version: 4.14.16
+      routes:
+        - name: Blackhole
+          address_prefix: 0.0.0.0/0
+          next_hop_type: none
+    udr_no_null414:
+      name: aro-414
+      version: 4.14.16
+      routes:
+        - name: To Internet
+          address_prefix: 0.0.0.0/0
+          next_hop_type: internet
+    udr413:
+      name: aro-413
+      version: 4.13.40
+      routes:
+        - name: Blackhole
+          address_prefix: 0.0.0.0/0
+          next_hop_type: none
+  vars:
+    resource_group: "{{ CLUSTERPREFIX }}-{{ inventory_hostname }}-{{ location }}"
+    network_prefix_cidr: 10.0.0.0/22
+    master_cidr: 10.0.0.0/23
+    worker_cidr: 10.0.2.0/23
+    apiserver_visibility: Private
+    ingress_visibility: Private
+    outbound_type: UserDefinedRouting
+
+upgrade_clusters:
+  hosts:
+    upgrade-412to416:
+      name: aro
+      version: 4.12.25
+      upgrade:
+        # https://access.redhat.com/labs/ocpupgradegraph/update_path?channel=stable-4.13&arch=x86_64&is_show_hot_fix=false&current_ocp_version=4.12.25&target_ocp_version=4.15.22
+        - from: "4.12"
+          channel: stable-4.13
+          version: 4.13.45
+          admin_acks:
+            - {"data":{"ack-4.12-kube-1.26-api-removals-in-4.13":"true"}}
+          allow-not-recommended: true
+        - from: "4.13"
+          channel: stable-4.14
+          version: 4.14.33
+          admin_acks:
+            - {"data":{"ack-4.13-kube-1.27-api-removals-in-4.14":"true"}}
+        - from: "4.14"
+          channel: stable-4.15
+          version: 4.15.22
+        - from: "4.15"
+          channel: stable-4.16
+          version: latest
```

Comment:
the stable-4.16 is empty right now, also an admin-ack is required, if really need to proceed like me, can go with candidate-4.16 instead:
```
        - from: "4.15"
          channel: candidate-4.16
          version: 4.16.5
          admin_acks:
            - {"data":{"ack-4.15-kube-1.29-api-removals-in-4.16":"true"}}

```